src/BlockchainReader/Bitcoin.ts

```diff
@@ -0,0 +1,53 @@
+import { PoetTimestamp, TransactionPoetTimestamp } from '@po.et/poet-js'
+
+import { PREFIX_BARD, PREFIX_POET, Block, Transaction, VOut } from 'Helpers/Bitcoin'
+
+interface VOutWithTxId extends VOut {
+  readonly transactionId: string
+}
+
+export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetTimestamp> =>
+  block.tx
+    .map(transactionToPoetAnchor)
+    .filter(_ => _)
+    .filter(poetAnchorHasCorrectPrefix)
```

Comment:
If this function could handle null entries the filter() above could be removed.
Perhaps transactionToPoetAnchor() should strip out any null records?

Reply:
In general I rather avoid adding null/undefined/empty input checking to functions so their scope and signature are more specific.

I'm thinking I could replace `.filter(_ => _)` with `.filter(identity)` (importing `identity` from Ramda), thought. What do you think?

Reply:
I don't think using `identity` would improve it. If it's a pattern you use alot, I'd extract it to a utility and give it a more meaningful name, like `stripNulls` or something.

Reply:
What about [this](https://github.com/poetapp/node/pull/383/commits/de1699c1a18fd44d03cd6f456b71f3066f097d7e)?

Reply:
That's fine too.